framework/src/modules/nft/commands/transfer.ts

```diff
@@ -43,31 +41,24 @@ export class TransferCommand extends BaseCommand {
 
 	public async verify(context: CommandVerifyContext<Params>): Promise<VerificationResult> {
 		const { params } = context;
+		const methodContext = context.getMethodContext();
 
-		const nftStore = this.stores.get(NFTStore);
-
-		const nftExists = await nftStore.has(context, params.nftID);
-
-		if (!nftExists) {
-			throw new Error('NFT substore entry does not exist');
+		let nft;
+		try {
+			nft = await this._method.getNFT(methodContext, params.nftID);
+		} catch (error) {
+			throw new Error('NFT does not exist');
```

Comment:
What is the reason the code is catching and re-throwing the error from `getNFT()` method?

`getNFT()` already throws 2 types of errors, depending on if the entry is missing from NFT store or from User store.

It seems that this try/catch not only complicates the command verification, but also provide less accurate error message than if there was no try/catch here.

Reply:
In this particular case (and verify hook of other commands), I think it makes sense to emit more granular error(s). And yes it'll get rid of the try-catch. Since this code will already be replaced by `verifyTransfer`, I'll apply the suggestion directly in the internal functions in PR #9005 instead of redundant update here.

Reply:
will update in the internal functions in PR #9005 to catch and throw if error is other than expected 👍